### DISCUSSION TITLE
inactive users

### DISCUSSION THREAD

**Discussion by Janus1974** [community/community#145842] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

I'm trying to find a way, by using python, to report any users that haven't accessed an org (or repo) in past 30 days.  I can retrieve a users contributions, but we have a number of management users who need access but don't contribute.

I guess what I'm looking for a way to find the last login date for a user.

I've been trying to find a way to get an inactive users list for days now, so any help is gratefully appreciated!

**Comment by checorazza** [comment, marked as answer]
Hey Janus1974!
Unfortunately, GitHub API doesn't provide an explicit "last login" endpoint, but you can infer activity by looking at:

- Contributions: This can include commits, pull requests, issues, etc.
- Repository Events: You can look at events like push and pull request activity.

You can try the following: 

1. Retrieve all users in your organization/repository:

2. Check Activity: For each user, check their contribution activity in the past 30 days. 
- You can use the /repos/:owner/:repo/commits endpoint to see if a user has committed in the past month
- Also you can use the /users/:username/events endpoint to check for any event activity.

3. Compare Last Activity Date: Check commit events or pull request creation dates for each user. If these were more than 30 days ago, they can be considered inactive.
